pkg/integrations/stub_integration.go

```diff
@@ -0,0 +1,73 @@
+package integrations
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/pkg/integrations/config"
+)
+
+// StubConfig represents a base no-op configuration, which returns a StubIntegration
+type StubConfig struct {
+	name   string
+	reason string
+	Common config.Common          `yaml:",inline"`
+	_      map[string]interface{} `yaml:"-"`
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for StubConfig
+func (c *StubConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = StubConfig{}
+	type plain StubConfig
+	err := unmarshal((*plain)(c))
+	return err
+}
```

Comment:
This works, in as much as it compiles, and is happy.

However, the desired behavior is that the config will still unmarshal, throwing away undefined values, and only log a warning when `StubConfig.NewIntegration` is called.

In practice, this isn't what happens, as we get yaml unmarshalling errors.

```
.\agent-windows-amd64.exe '-config.file' cadvisorcfg.yml
2021-12-14 16:37:01.286665 I | error loading config file cadvisorcfg.yml: yaml: unmarshal errors:
  line 4: field containerd_namespace not found in type integrations.plain
  line 5: field disabled_metrics not found in type integrations.plain
```

Ideally, I'd like to sort that out. 🤔

Reply:
Resolved by not having a `StubConfig` see [this conversation](https://github.com/grafana/agent/pull/1081#discussion_r769162251)